src/sentry/models/commitfilechange.py

```diff
@@ -48,17 +48,20 @@ def process_resource_change(instance, **kwargs):
     from sentry.integrations.gitlab.integration import GitlabIntegration
     from sentry.tasks.codeowners import code_owners_auto_sync
 
-    filepaths = set(GitHubIntegration.codeowners_locations) | set(
-        GitlabIntegration.codeowners_locations
-    )
-
-    # CODEOWNERS file added or modified, trigger auto-sync
-    if instance.filename in filepaths and instance.type in ["A", "M"]:
-        # Trigger the task after 5min to make sure all records in the transactions has been saved.
-        code_owners_auto_sync.apply_async(
-            kwargs={"commit_id": instance.commit_id}, countdown=60 * 5
+    def _spawn_task():
+        filepaths = set(GitHubIntegration.codeowners_locations) | set(
+            GitlabIntegration.codeowners_locations
         )
 
+        # CODEOWNERS file added or modified, trigger auto-sync
+        if instance.filename in filepaths and instance.type in ["A", "M"]:
+            # Trigger the task after 5min to make sure all records in the transactions has been saved.
+            code_owners_auto_sync.apply_async(
+                kwargs={"commit_id": instance.commit_id}, countdown=60 * 5
+            )
```

Comment:
This delay might be redundant now, but I understand that poking that bear is likely out of scope for this change.

Reply:
i didn't even notice. yeah probably but i'm not keen on finding out